### ISSUE TITLE
[Bug]: Cannot stream with ffmpeg from doorbell when running with TrueNAS FreeBSD.

### ISSUE THREAD

**Issue by zthun** [homebridge-eufy-security/plugin#123]
### What happened?

> Note:  I'm not sure if this is an ffmpeg problem or an issue with some command line argument but logs are attached.  I also know that my OS here is unsupported, but I figured, worst case scenario, I can fork and try to fix myself with a little guidance.  I did also check the Common issues page and the workaround proposed didn't work. 

**Environment**:
1.  TrueNAS Core @2021
2. OS: FreeBSD 12.2-RELEASE-p10 b26f74b5984(HEAD) TRUENAS

Note that the OS is running in a Jail 12.3-RELEASE.  

**Repro Steps:**

1.  Using  v2.1.0-beta.12 
4. Eufy Doorbell (https://www.amazon.com/gp/product/B084MFPRCP/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1)
5. Open homekit
6. Click on Camera for doorbell in appropriate room.

**Expected Output:**

* Camera streams successfully to homekit

**Actual Output**

* Camera gets a not responding error or spins forever (inconsistent).  

### Device Type

Doorbell (Supported)

### Plugin Version

beta (Best Effort)

### HomeBridge Version

v1.4.x (Supported)

### NodeJS Version

v16 (Supported)

### Operating System

Other (UnSupported)

### Relevant log output

```shell
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 30 from stream 0 at ts 28
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [info] frame=   31 fps=0.0 q=47.0 size=      39kB time=00:00:01.00 bitrate= 317.0kbits/s dup=12 drop=3 speed= 1.7x    
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 33 from stream 0 at ts 30
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 33 from stream 0 at ts 30
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 33 from stream 0 at ts 31
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 2 dup!
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [info] frame=   39 fps= 34 q=40.0 size=      48kB time=00:00:01.26 bitrate= 307.3kbits/s dup=15 drop=6 speed=1.11x    
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 2 dup!
[7/23/2022, 12:21:07 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 2 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 2 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [info] frame=   52 fps= 25 q=40.0 size=      57kB time=00:00:01.70 bitrate= 276.6kbits/s dup=21 drop=6 speed=0.804x    
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 2 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:08 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 2 dup!
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 14 dup!
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [info] frame=   94 fps= 36 q=29.0 size=     104kB time=00:00:03.10 bitrate= 274.2kbits/s dup=49 drop=6 speed=1.18x    
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 94 from stream 0 at ts 92
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 94 from stream 0 at ts 92
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: handleSnapshotRequest
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: snapshotPromise: false
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Snapshot requested: 1280 x 720 Vision true
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Locked refresh process.
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Fetching new snapshot from camera.
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision RTSP rtspStream:false
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Looks like not compatible with RTSP
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision New instance requests livestream. There were 1 instance(s) using the livestream until now.
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Using livestream that was started 3.862 seconds ago. The proxy stream has id: 6.
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: current_path /root/.homebridge/eufysecurity/20.sock
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: parent_dir /usr/local/lib/node_modules/homebridge-eufy-security/dist/accessories
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Snapshot command: ffmpeg -i unix:/root/.homebridge/eufysecurity/20.sock -frames:v 1 -f image2 - -hide_banner -loglevel error true
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 95 from stream 0 at ts 92
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 95 from stream 0 at ts 92
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] Past duration 0.647789 too large
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** dropping frame 95 from stream 0 at ts 93
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 4 dup!
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [info] frame=  103 fps= 33 q=27.0 size=     112kB time=00:00:03.40 bitrate= 270.4kbits/s dup=54 drop=11 speed=1.07x    
[7/23/2022, 12:21:09 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] Past duration 0.830269 too large
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [info] frame=  115 fps= 31 q=31.0 size=     124kB time=00:00:03.80 bitrate= 266.6kbits/s dup=58 drop=11 speed=1.02x    
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 2 dup!
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: snapshot byte lenght: 103405
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Sending snapshot: 1280 x 720 Vision true
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Resize command: ffmpeg -i pipe: -frames:v 1 -filter:v scale='min(1280,iw)':'min(720,iw)':force_original_aspect_ratio=decrease -f image2 - true
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: [Vision] [Video Process] [verbose] *** 1 dup!
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Receive Apple HK Stop request{"sessionID":"f761420b-35ba-4572-a41b-85ef13db4c95","type":"stop"}
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Stopping session with id: f761420b-35ba-4572-a41b-85ef13db4c95
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision RTSP rtspStream:false
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Looks like not compatible with RTSP
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision_video Closed NamePipeStream
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision_audio Closed NamePipeStream
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision RTSP rtspStream:false
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Looks like not compatible with RTSP
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Audiostream was stopped after transmission of 0 data chunks.
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Videostream was stopped after transmission of 83 data chunks.
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision One stream instance (id: 5) released livestream. There are now 1 instance(s) using the livestream.
[7/23/2022, 12:21:10 PM] [EufySecurity-2.1.0-beta.12] INFO:  Vision Stopped video stream.
[7/23/2022, 12:21:12 PM] [EufySecurity-2.1.0-beta.12] INFO:  [Vision] [Video Process] FFmpeg exited with code: null and signal: SIGKILL (Forced)
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Audiostream was stopped after transmission of 0 data chunks.
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Videostream was stopped after transmission of 79 data chunks.
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision One stream instance (id: 6) released livestream. There are now 0 instance(s) using the livestream.
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision All proxy instances to the livestream have terminated.
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Stopping station livestream.
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Fetching snapshot took 4.853 seconds. Vision true
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision Unlocked refresh process.
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] DEBUG: Vision store new snapshot from camera in memory. Using this for future use.
[7/23/2022, 12:21:14 PM] [EufySecurity-2.1.0-beta.12] INFO:  Oracle station livestream for Vision has stopped.
```

**Comment by thieren** [comment]
Hi @zthun 

Sure we can try to get this working for you.

Firstly I like you to share your config json. That will make it a lot easier to see what's happening in the logs.
I have already seen that you're using camera snapshots. To minimise the side effects I'd suggest that you revert to cloud snapshots and also disable audio and talkback, if that's enabled for you.
This will ensure that there is only one ffmpeg process running for streaming.

If that doesn't help (it won't I guess, but debugging will be easier) you could send me a full log, where all the background communication is logged. In the latest beta is a download log option inside the plugin settings.
You could send this to me via discord.

**Comment by zthun** [comment]
So interestingly enough, I tested a few more options.  

I turned off "Use Enhanced Snapshot Behavior" completely and then I got a stream.  This was odd.  I then turned it back on and put it to cloud snapshots.  The behavior returned to just spinning forever.  Turned it back off, and then I could get the stream again.  

I then tried turning on the deprecated "Force Refresh Snapshot (Deprecated)" to see if it was related just retrieving snapshots completely, and sure enough, it returned to just spinning forever.  

Somehow, having snapshots enabled just kills the stream.
